Libraries/dotNetRDF/Writing/Formatting/BaseFormatter.cs

```diff
@@ -118,9 +118,7 @@ public virtual String Format(Triple t)
         /// <returns></returns>
         public virtual String FormatUri(String u)
         {
-            // String uri = Uri.EscapeUriString(u);
-            u = u.Replace(">", "\\>");
-            return u;
+            return Rfc3987Formatter.EscapeUriString(u);
```

Comment:
Interesting how it was `Uri.EscapeUriString` originally.

What was the reason actually to implement the `Rfc3987Formatter`. `System.Uri` is not conformant?

Reply:
IIRC there were some cross-platform issues. .NET Core and later .NET Framework properly implement IRI support. Earlier versions didn't. Looking at the docs (https://docs.microsoft.com/en-us/dotnet/api/system.uri?view=netcore-3.1), this appears to have been fixed - though .NET 4.0 users might have to fiddle with app.config to enable IRI support.

Might be worth trying to remove this formatter class - it's not 100% correct in its approach in any case and it would be less code to manage.

Reply:
The tests are passing on net452. Are we even going to support older versions? I think it hardly makes sense even for <net462, correct?

> it's not 100% correct in its approach in any case and it would be less code to manage.

Maybe it is correct now? In any case, less code is always better

Reply:
Ah, no luck with `Uri.EscapeUriString'. It suffers from the same problem if presented with a partially escaped string

```
// will also double-escape
Uri.EscapeUriString("http://example.com/Person/2/Bob/September%2C 2010")

// will leave coma (reserved char)
Uri.EscapeUriString("http://example.com/Person/2/Bob/September, 2010") 
```

Reply:
Ah well, it was worth a try, thanks :-)

Yeah support for older versions of .NET is definitely something to discuss - but probably a bigger issue than this PR!

Reply:
Aha, the main binary is indeed compiled to `net40`. Then maybe unit tests should also run against that platform?

Reply:
Yeah, ideally it would. I just worry that adding a third target framework to the unit tests will cause us to hit Appveyor build time limits.

Reply:
I have been quite happy with GitHub Actions in many of my repos. They are very fast (definitely faster then Travis, have not enough to compare with Appveyor)

And a big plus is ability to spread a build across multiple jobs which run in parallel. For example a job per target

Reply:
Created #330 to remind myself to take a look at some point!